kroxylicious/example-proxy-config.yml

```diff
@@ -9,6 +9,9 @@ proxy:
   address: localhost:9192
   logNetwork: true
   logFrames: true
+metrics:
```

Comment:
nit: should maybe make it explicit that its prometheus metrics

Reply:
it'd be good to have the prometheus noun in there somewhere, but maybe under `metrics`, in case we wanted to support different backends.

Like:

```
metrics:
    host: localhost
    port: 1234
    prometheus:
        enabled: true
```

Reply:
yeah, maybe needs to be more like 
```yaml
metrics:
    host: localhost
    port: 1234
    prometheus:
        enabled: true
        path: /metrics
  ```
  
  But I think the general point is valid. I'm on the fence about having an enabled flag, if you have metrics configured surely you want them enabled? Or at least default it to on if configured...

Reply:
I can imagine if you had accumulated a lot of configuration under prometheus (like if we added configuration for histogram precision) and wanted to switch it off for whatever reason an enabled flag is handy rather than removing a whole block.

Yep, default enabled sound good. We could support `prometheus: {}` as enabled.

Reply:
Have done the above and added the `path` support, it defaults to `/metrics` and 404s on anything else.

Reply:
Another idea is to split metrics config from "management-endpoint" configuration, in case we want more that metrics on it. (dynamic filter reconfiguration?) Thinking along the lines of micronaut's [config style](https://micronaut-projects.github.io/micronaut-micrometer/latest/guide/#metricsAndReportersPrometheus)

Reply:
I don't think we want an `enabled` flag: They way lies madness because you could argue that any particular feature might need to be disabled, so everything ends up with the flag. If you want to switch it off you can use YAML `#` comments to just temporarily remove the config.

Reply:
With this configuration we're kind of saying that this "HTTP server" can only be used for metrics (albeit possibly for a number of different metric formats). I suspect we might end up with other use cases for getting information out of (or in to) a proxy instance. So perhaps we should think of this in terms of providing a path/endpoint in a possibly-shared HTTP server which might also be used to host other non-metric endpoints. I.e. I think we might come to regret putting the `host` and `port` below `metrics`.

Reply:
I've refactored it so that the top level configuration is `adminHttp` and it's now configured like:

```
adminHttp:
  host: localhost
  port: 9193
  endpoints:
    prometheus: {}
```
Along with refactoring to introduce a `RoutingHttpServer` to handle different endpoints.

Any advances on the `adminHttp` naming is welcome.